openedx/event-routing-backends
# Emit multiple xAPI events for a multi-question submission

**Description:**

Updates the base event processor and tests to support transforming a single event into multiple events, and
modifies the server-side `problem_check`  xAPI event transformer to transform multi-problem submissions into:

* 1 event for the problem submission, plus
* 1 event for each sub-problem

Resolves https://github.com/openedx/event-routing-backends/issues/219

**JIRA:** [FAL-3430](https://tasks.opencraft.com/browse/FAL-3430) (OpenCraft internal link)

**Testing instructions:**

This change should be covered by tests, but to test it manually:

1. Install this branch on your Tutor LMS
2. Login, and submit various responses to the [Multiple Choice Questions](http://local.overhang.io:2000/course/course-v1:OpenCraft+DemoX+Demo/block-v1:OpenCraft+DemoX+Demo+type@sequential+block@basic_questions/block-v1:OpenCraft+DemoX+Demo+type@vertical+block@54bb9b142c6c4c22afc62bcb628f0e68) block in the Demo course, or create your own.
3. Watch multiple xAPI events flow through into Aspects from a single submission.

**Merge checklist:**

**Post merge:**
      finished.

**Author concerns:** 

* Suggest reviewing this commit by commit to understand the reasoning.
* This change only affects xAPI, not Caliper events.
* Some minor changes to single question problem_check events occurred: added `object.definition.name` and `interactionType`.
